helix-term/src/commands/typed.rs

```diff
@@ -2408,6 +2408,57 @@ fn redraw(
     Ok(())
 }
 
+fn move_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    ensure!(args.len() == 1, format!(":move takes one argument"));
+
+    let new_path = args.first().unwrap().to_string();
+    let doc = doc!(cx.editor);
+    let old_path = doc
+        .path()
+        .ok_or_else(|| anyhow!("Scratch buffer cannot be moved. Use :write instead"))?
+        .to_string_lossy()
```

Comment:
Lossy conversion is incorrect. If there was no way to recover here, we should report an error to the user.

Luckily its perfectly possible to construct URLs for invalid utf8 (by percent-encoding non-ascii bytes). So if you use `document.url` (and accept that as arguments for will/did_rename) then that will cover all cases.

This will also prefix the path with `file://` (the fact that it doesn't currently do that is another bug here, the spec specifies that this should be a file:// URL and rust-analyzer also implements it like that)

Reply:
Updated :) I struggled while dealing between Url and Path (as I need them on on the other side) but I think I got something okay.
Let me know if you see any improvements !

Reply:
no, worries it looks good now